apps/drec-api/src/pods/organization/organization.service.ts

```diff
@@ -212,6 +235,60 @@ export class OrganizationService {
     await this.sendRoleChangeEmail(organization, userToBeChanged, newRole);
   }
 
+  async setBlockchainAddress(
+    id: number,
+    signedMessage: BindBlockchainAccountDTO['signedMessage'],
+  ): Promise<ISuccessResponse> {
+    if (!signedMessage) {
+      throw new BadRequestException('Signed message is empty.');
+    }
+
+    const organization = await this.findOne(id);
+
+    if (organization.blockchainAccountAddress) {
+      throw new ConflictException(
+        'Organization already has a blockchain address',
+      );
+    }
+
+    const address = await recoverTypedSignatureAddress(
+      this.configService.get<string>('REGISTRATION_MESSAGE_TO_SIGN') ||
```

Comment:
Shouldn't this message to sign be unique for each user? Like contain some user id or any kind of salt?

Reply:
Based on the approach from Origin, it is not. It is just set on the environment